pkg/trace/stats/span_concentrator.go

```diff
@@ -9,19 +9,106 @@ import (
 	"sync"
 	"time"
 
+	"github.com/DataDog/datadog-agent/pkg/obfuscate"
 	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
 	"github.com/DataDog/datadog-agent/pkg/trace/log"
-	"github.com/DataDog/datadog-agent/pkg/trace/traceutil"
 )
 
+// This is a special metric, it's 1 if the span is top-level, 0 if not.
+const topLevelKey = "_top_level"
+
+// measuredKey is a special metric flag that marks a span for trace metrics calculation.
+const measuredKey = "_dd.measured"
+
+// partialVersionKey is a metric carrying the snapshot seq number in the case the span is a partial snapshot
+const partialVersionKey = "_dd.partial_version"
+
 // SpanConcentratorConfig exposes configuration options for a SpanConcentrator
 type SpanConcentratorConfig struct {
 	// ComputeStatsBySpanKind enables/disables the computing of stats based on a span's `span.kind` field
 	ComputeStatsBySpanKind bool
 	// BucketInterval the size of our pre-aggregation per bucket
 	BucketInterval int64
-	// PeerTags additional tags to use for peer entity stats aggregation, nil if disabled
-	PeerTags []string
+}
+
+// StatSpan holds all the required fields from a span needed to calculate stats
+type StatSpan struct {
+	service  string
+	resource string
```

Comment:
I explicitly avoided implementing and using an interface to avoid the performance implications of turning struct field accesses into function calls (which because of interfaces won't be able to be inlined)